packages/react/src/components/Popup/Popup.tsx

```diff
@@ -245,46 +223,30 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     )
   }
 
-  private updateOutsideHandleSubscription() {
-    if (this.state.open && this.outsideClickSubscription.isEmpty) {
-      setTimeout(() => {
-        this.outsideClickSubscription = EventStack.subscribe(
-          'click',
-          e => {
-            if (this.isOutsidePopupElementAndOutsideTriggerElement(e)) {
-              this.state.open && this.trySetOpen(false, e)
-            }
-          },
-          {
-            useCapture: true,
-          },
-        )
-        this.outsideKeySubscription = EventStack.subscribe(
-          'keydown',
-          e => {
-            const keyCode = keyboardKey.getCode(e)
-
-            const matchingKey = keyCode === keyboardKey.Enter || keyboardKey.Spacebar
-
-            if (matchingKey && this.isOutsidePopupElementAndOutsideTriggerElement(e)) {
-              this.state.open && this.trySetOpen(false, e)
-            }
-          },
-          {
-            useCapture: true,
-          },
-        )
-      })
-    } else {
-      this.outsideClickSubscription.unsubscribe()
-      this.outsideKeySubscription.unsubscribe()
+  handleDocumentClick = (getRefs: Function) => e => {
+    if (this.isOutsidePopupElementAndOutsideTriggerElement(getRefs(), e)) {
+      this.trySetOpen(false, e)
     }
   }
 
-  private isOutsidePopupElementAndOutsideTriggerElement(e) {
-    const isOutsidePopupElement = this.popupDomElement && !this.popupDomElement.contains(e.target)
+  handleDocumentKeyDown = (getRefs: Function) => (e: KeyboardEvent) => {
+    const keyCode = keyboardKey.getCode(e)
+    const isMatchingKey = keyCode === keyboardKey.Enter || keyCode === keyboardKey.Spacebar
+
+    if (isMatchingKey && this.isOutsidePopupElementAndOutsideTriggerElement(getRefs(), e)) {
+      this.trySetOpen(false, e)
+    }
+  }
+
+  private isOutsidePopupElementAndOutsideTriggerElement(refs: NodeRef[], e) {
```

Comment:
nit: might suggest to remove `private` modifier for the sake of consistency with other members

Reply:
Removed all visibility modifiers in `Popup.tsx` 👍